tests/utils/tensorflow/test_rasa_layers.py

```diff
@@ -0,0 +1,855 @@
+            {"attribute_signature": attribute_signature_basic},
+            attribute_features_basic,
+            [
+                [batch_size, max_seq_length + 1, units_hidden_layer],
+                [batch_size, max_seq_length + 1, units_hidden_layer],
+                [batch_size, max_seq_length + 1, 1],
+                [0,],
+                [0,],
+                [0,],
+            ],
+            "same_as_train",
+        ),
+        # Sequence layer with mixed features, no hidden layers, no transformer, no MLM
+        (
+            RasaSequenceLayer,
+            model_config_basic_no_hidden_layers,
+            {"attribute_signature": attribute_signature_basic},
+            attribute_features_basic,
+            [
+                [batch_size, max_seq_length + 1, units_concat],
+                [batch_size, max_seq_length + 1, units_concat],
+                [batch_size, max_seq_length + 1, 1],
+                [0,],
+                [0,],
+                [0,],
+            ],
+            "same_as_train",
+        ),
+        # Sequence layer with only sequence-level sparse features & MLM (to check the
+        # shape of token_ids)
+        (
+            RasaSequenceLayer,
+            model_config_transformer_mlm,
+            {
+                "attribute_signature": {
+                    SEQUENCE: [feature_signature_sparse_1],
+                    SENTENCE: [],
+                }
+            },
+            ([feature_sparse_seq_1], [], sequence_lengths,),
+            [
+                [batch_size, max_seq_length, units_transformer],
+                [batch_size, max_seq_length, units_hidden_layer],
+                [batch_size, max_seq_length, 1],
+                [batch_size, max_seq_length, 2],
+                [batch_size, max_seq_length, 1],
+                [
+                    num_transformer_layers,
+                    batch_size,
+                    num_transformer_heads,
+                    max_seq_length,
+                    max_seq_length,
+                ],
+            ],
+            [
+                [batch_size, max_seq_length, units_transformer],
+                [batch_size, max_seq_length, units_hidden_layer],
+                [batch_size, max_seq_length, 1],
+                [0,],
+                [0,],
+                [
+                    num_transformer_layers,
+                    batch_size,
+                    num_transformer_heads,
+                    max_seq_length,
+                    max_seq_length,
+                ],
+            ],
+        ),
+    ],
+)
+def test_correct_output_shape(
+    layer_class: Callable[..., Any],
+    model_config: Dict[Text, Any],
+    layer_args: Dict[Text, Any],
+    layer_inputs: List[  # TODO really use List? Should be Tuple of variable length...
+        List[Union[tf.SparseTensor, tf.Tensor]]
+    ],
+    expected_output_shapes_train: List[List[int]],
+    expected_output_shapes_test: Union[Text, List[Union[int, List[int]]]],
```

Comment:
`expected_output_shapes_test` can have annotation `Union[Text, List[List[int]]]` can't it?

Reply:
Good catch, I think the annotation was needed like these previously but today it can be simplified as you're suggesting.